jenkins-integration.sh

```diff
@@ -0,0 +1,162 @@
+  # CE
+  go test -coverprofile=coverage_walrus_ce.out -coverpkg=github.com/couchbase/sync_gateway/${TARGET_PACKAGE} $GO_TEST_FLAGS github.com/couchbase/sync_gateway/${TARGET_PACKAGE} >verbose_unit_ce.out.raw 2>&1 | true
+fi
+
+# Start CBS
+docker stop couchbase || true
+docker rm couchbase || true
+# -p 8091-8094:8091-8094 -p 11210:11210
+docker run -d --name couchbase --net=host couchbase/server:${COUCHBASE_SERVER_VERSION}
+
+# Test to see if Couchbase Server is up
+# Each retry min wait 5s, max 10s. Retry 20 times with exponential backoff (delay 0), fail at 120s
+curl --retry-all-errors --connect-timeout 5 --max-time 10 --retry 20 --retry-delay 0 --retry-max-time 120 'http://127.0.0.1:8091'
+
+# Set up CBS
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/nodes/self/controller/settings -d 'path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&' -d 'index_path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&' -d 'cbas_path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&' -d 'eventing_path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/node/controller/rename -d 'hostname=127.0.0.1'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/node/controller/setupServices -d 'services=kv%2Cn1ql%2Cindex'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/pools/default -d 'memoryQuota=3072' -d 'indexMemoryQuota=3072' -d 'ftsMemoryQuota=256'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/settings/web -d 'password=password&username=Administrator&port=SAME'
+curl -u Administrator:password -v -X POST http://localhost:8091/settings/indexes -d indexerThreads=4 -d logLevel=verbose -d maxRollbackPoints=10 \
+-d storageMode=plasma -d memorySnapshotInterval=150 -d stableSnapshotInterval=40000
+
+sleep 10
+
+# Set up test environment variables for CBS runs
+export SG_TEST_USE_XATTRS=${XATTRS}
+export SG_TEST_USE_GSI=${GSI}
+export SG_TEST_COUCHBASE_SERVER_URL="${COUCHBASE_SERVER_PROTOCOL}://127.0.0.1"
+export SG_TEST_BACKING_STORE="Couchbase"
+export SG_TEST_BUCKET_POOL_SIZE=${SG_TEST_BUCKET_POOL_SIZE}
+export SG_TEST_TLS_SKIP_VERIFY=${TLS_SKIP_VERIFY}
+
+if [ "${SG_EDITION}" == "EE" ]; then
+    GO_TEST_FLAGS="${GO_TEST_FLAGS} -tags cb_sg_enterprise"
+fi
+
+go test ${GO_TEST_FLAGS} -coverprofile=coverage_int.out -coverpkg=github.com/couchbase/sync_gateway/${TARGET_PACKAGE} github.com/couchbase/sync_gateway/${TARGET_PACKAGE} 2>&1 | tee "${INT_LOG_FILE_NAME}.txt" | grep -E '(--- (FAIL|PASS|SKIP):|github.com/couchbase/sync_gateway(/.+)?\t|TEST: |panic: )'
+if [ "${PIPESTATUS[0]}" -ne "0" ]; then # If test exit code is not 0 (failed)
+  echo "Go test failed! Parsing logs to find cause..."
+  TEST_FAILED=true
+fi
+
```

Comment:
Missing in this script vs. what we used to run for integration tests:

```
# Strip non-printable characters
LC_CTYPE=C tr -dc [:print:][:space:] < verbose_unit_ee.out.raw > verbose_unit_ee.out
LC_CTYPE=C tr -dc [:print:][:space:] < verbose_unit_ce.out.raw > verbose_unit_ce.out
LC_CTYPE=C tr -dc [:print:][:space:] < verbose_int.out.raw > verbose_int.out
```

This was used because some of our test logging can in some cases emit non-printable/non-valid UTF-8 characters from logging binary attachment data, which caused issues with creating valid XML.